### PR TITLE
Add async site connection to missing connection Notice

### PR DESCRIPTION
**Pull request by IanRamosC**
See linked issue

## Proposed changes:

* This PR updates the site connection button to perform an async site connection, instead of directing to the user connection screen. After a user clicks on the button, the button will get into a loading state and the site will try to connect to WordPress.com in the background. After the connection, it will add a confirmation notice, stating that the site has been successfully connected.

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-qKY-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Spin up a JN site with Jetpack Beta pointing to this branch
* Visit `/wp-admin/admin.php?page=my-jetpack`
* Dismiss the Welcome banner and refresh the page
* You'll now see a banner saying "Some products need a connection to WordPress.com to be able to work.". Click on "Connect your site to fix this"
* The label will change to the loading icon
* After that loading, you'll see a success notice with the following message: "Your site has been successfully connected."

NOTE: error handling and the state of the cards will be handled in a different PR

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-site-connection-notice.ts [hunk 1]
```diff
@@ -39,23 +79,47 @@ const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
 						'jetpack-my-jetpack'
 				  );
 
+		const needsSiteConnectionMessage = __(
+			'Some products need a connection to WordPress.com to be able to work.',
+			'jetpack-my-jetpack'
+		);
```
Comment (jboland88) [inline projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-site-connection-notice.ts]: With the site connection message - we need a way to include our TOS copy on the banner to indicate that clicking on the connection button is agreeing to our TOS.

We discussed handling that in another small PR - let's merge that together with this one before it goes into trunk.

File: projects/packages/my-jetpack/_inc/context/notices/types.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export type Notice = {
 		actions?: {
 			label: string;
 			onClick: () => void;
+			isLoading?: boolean;
```
Comment (jboland88) [inline projects/packages/my-jetpack/_inc/context/notices/types.ts]: Now that we have control of the notice component, could we customize the loading state to say "Connecting...." for this action? 

Maybe a custom loading text prop?

File: projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-site-connection-notice.ts [hunk 3]
```diff
@@ -39,23 +79,47 @@ const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
...
 			],
 		};
 
 		setNotice( {
-			message: needsUserConnectionMessage,
+			message: requiresUserConnection ? needsUserConnectionMessage : needsSiteConnectionMessage,
 			options: noticeOptions,
-			priority: NOTICE_PRIORITY_HIGH,
+			// If this notice gets into a loading state, we want to show it above the rest
+			priority: NOTICE_PRIORITY_HIGH + ( siteIsRegistering ? 1 : 0 ),
```
Comment (jboland88) [inline projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-site-connection-notice.ts]: Nice touch here.

**Review by jboland88 — CHANGES_REQUESTED** [review]
This tested well and is looking good - I really like how it works.

I think we need to add the TOS copy to the connection banner before we merge this.
Other feedback is non-blocking.

**Comment by IanRamosC** [comment]
Closing in favor of #36771

**Pull request closed without merging**